### PR TITLE
[ODS-6288] Rebase Epic branch (ODS-6105)

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: Application/EdFi.Ods.Common/Exceptions/ApiSecurityConfigurationException.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Common/Exceptions/ApiSecurityConfigurationException.cs:1]: This file was removed in ODS-6139

File: Application/EdFi.Ods.Common/Models/Resource/ProfileContentTypeException.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Common/Models/Resource/ProfileContentTypeException.cs:1]: This file was removed in ODS-6139

File: Application/EdFi.Ods.Common/Exceptions/DistributedCacheException.cs [hunk 3]
```diff
@@ -25,5 +24,13 @@ public class SafeDistributedCacheException : DistributedCacheException
 
         public SafeDistributedCacheException(string message)
             : base(message) { }
+
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Common/Exceptions/DistributedCacheException.cs:27]: I don't see this conde in ODS-6139 commit, was it added as part of the migration to .net 8?

File: Application/EdFi.Ods.Common/Security/EdFiSecurityConflictException.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+﻿// SPDX-License-Identifier: Apache-2.0
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Common/Security/EdFiSecurityConflictException.cs:1]: This file was removed in ODS-6139

File: Application/EdFi.Ods.Common/Security/EdFiSecurityException.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
+﻿// SPDX-License-Identifier: Apache-2.0
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Common/Security/EdFiSecurityException.cs:1]: This file was removed in ODS-6139

File: Application/EdFi.Ods.Features/Notifications/Redis/InitializeRedisNotifications.cs [hunk 6]
```diff
@@ -32,6 +32,7 @@ public class InitializeRedisNotifications : IExternalTask
         _notificationsMessageSink = notificationsMessageSink;
     }
 
+    [Obsolete]
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Features/Notifications/Redis/InitializeRedisNotifications.cs:35]: Is this needed to supress a warning? I don't remember it being as part of one of the .NET 8 migration tickets

**Review by simpat-jesus — CHANGES_REQUESTED** [review]

**Comment by simpat-jesus** [comment]
We also need to remove the customizations pulled from .NET 8 for tickets [ODS-6185](https://tracker.ed-fi.org/browse/ODS-6185) and [ODS-6139](https://tracker.ed-fi.org/browse/ODS-6139)

**Review by simpat-jesus — APPROVED** [review]

**Review by simpat-jesus — APPROVED** [review]

**Review by vimayya — APPROVED** [review]

**Pull request merged**